lib/rules/no-constant-condition.js

```diff
@@ -19,10 +19,12 @@ module.exports = function(context) {
     /**
      * Checks if a node has a constant truthiness value.
      * @param {ASTNode} node The AST node to check.
+     * @param {boolean} inBooleanPosition `false` if checking branch of a condition.
```

Comment:
"boolean position" means it is in a position where, regardless what it returns, only the truthiness of the return value will matter.